java/maven/src/org/netbeans/modules/maven/InternalActionProvider.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.netbeans.modules.maven;
+
+import org.netbeans.spi.project.ActionProvider;
+
+/**
+ * The same as ActionProvider, but won't be exported outside the module.
+ * @author sdedic
+ */
+public interface InternalActionProvider extends ActionProvider {
```

Comment:
Anyone who queries `Lookup.lookupAll(ActionProvider.class)` may get instances of `InternalActionProvider`. Is that intended? If not, I suggest to use delegation:
```java
public interface InternalActionProvider {
  ActionProvider getProvider();
}
```

Reply:
Ok, done in 04297683b3

Reply:
Still a bit [dangerous](https://github.com/apache/netbeans/pull/2795#commitcomment-47981828).